addons/test_discuss_full/tests/test_performance.py

```diff
@@ -1765,3 +1765,32 @@ def _expected_result_for_thread(self, channel):
         if channel == self.channel_livechat_2:
             return {**common_data, "display_name": "anon 2 Ernest Employee"}
         return {}
+
+
+@tagged("post_install", "-at_install", "is_query_count")
+class TestRatingStatsPerformance(HttpCase, MailCommon):
+    @warmup
+    def test_fetch_rating_stats(self):
+        """
+        Computation of rating_stats should run a single query per model with rating_stats enabled.
+        """
+        record_a = self.env["mail.test.rating"].create({"name": "Ticket A1"})
+        record_b = self.env["slide.channel"].create({"name": "Course B1"})
+
+        for record in [record_a, record_b]:
+            record.message_post(
+                body=f"<p>Test message for {record.name}</p>",
+                message_type="comment",
+                partner_ids=[self.user_employee.partner_id.id],
+                rating_value="4",
+            )
+        self.authenticate(self.user_employee.login, self.user_employee.password)
+        self.env.registry.clear_cache()
+        self.env.invalidate_all()
```

Comment:
I suggest you to swap those two statements.
That's because `invalidate_all()` will recompute some fields, which may access some `ormcache`'d methods, hence voiding the effect of `registry.clear_cache()`.